include/deal.II/simplex/barycentric_polynomials.h

```diff
@@ -0,0 +1,639 @@
+     */
+    static BarycentricPolynomials<dim>
+    get_fe_p_basis(const unsigned int degree);
+
+    /*
+     * Constructor taking the polynomial @p degree as input.
+     */
+    BarycentricPolynomials(
+      const std::vector<BarycentricPolynomial<dim>> &polynomials);
+
+    /**
+     * @copydoc ScalarPolynomialsBase::evaluate()
+     */
+    void
+    evaluate(const Point<dim> &           unit_point,
+             std::vector<double> &        values,
+             std::vector<Tensor<1, dim>> &grads,
+             std::vector<Tensor<2, dim>> &grad_grads,
+             std::vector<Tensor<3, dim>> &third_derivatives,
+             std::vector<Tensor<4, dim>> &fourth_derivatives) const override;
+
+    /**
+     * @copydoc ScalarPolynomialsBase::compute_value()
+     */
+    double
+    compute_value(const unsigned int i, const Point<dim> &p) const override;
+
+    /**
+     * @copydoc ScalarPolynomialsBase::compute_1st_derivative()
+     */
+    Tensor<1, dim>
+    compute_1st_derivative(const unsigned int i,
+                           const Point<dim> & p) const override;
+
+    /**
+     * @copydoc ScalarPolynomialsBase::compute_2nd_derivative()
+     */
+    Tensor<2, dim>
+    compute_2nd_derivative(const unsigned int i,
+                           const Point<dim> & p) const override;
+
+    /**
+     * @copydoc ScalarPolynomialsBase::compute_3rd_derivative()
+     */
+    Tensor<3, dim>
+    compute_3rd_derivative(const unsigned int i,
+                           const Point<dim> & p) const override;
+
+    /**
+     * @copydoc ScalarPolynomialsBase::compute_4th_derivative()
+     */
+    Tensor<4, dim>
+    compute_4th_derivative(const unsigned int i,
+                           const Point<dim> & p) const override;
+
+    /**
+     * @copydoc ScalarPolynomialsBase::compute_grad()
+     */
+    Tensor<1, dim>
+    compute_grad(const unsigned int i, const Point<dim> &p) const override;
+
+    /**
+     * @copydoc ScalarPolynomialsBase::compute_grad_grad()
+     */
+    Tensor<2, dim>
+    compute_grad_grad(const unsigned int i, const Point<dim> &p) const override;
+
+    /**
+     * @copydoc ScalarPolynomialsBase::memory_consumption().
+     */
+    virtual std::size_t
+    memory_consumption() const override;
+
+    /**
+     * @copydoc ScalarPolynomialsBase::name()
+     */
+    std::string
+    name() const override;
+
+    /**
+     * @copydoc ScalarPolynomialsBase::clone()
+     */
+    virtual std::unique_ptr<ScalarPolynomialsBase<dim>>
+    clone() const override;
+
+  protected:
+    std::vector<BarycentricPolynomial<dim>> polys;
+
+    Table<2, BarycentricPolynomial<dim>> poly_grads;
+
+    Table<3, BarycentricPolynomial<dim>> poly_hessians;
+
+    Table<4, BarycentricPolynomial<dim>> poly_third_derivatives;
+
+    Table<5, BarycentricPolynomial<dim>> poly_fourth_derivatives;
+  };
+} // namespace Simplex
+
+// free template functions (in Simplex, not dealii::Simplex)
```

Comment:
I don't get this comment.

Reply:
It's wrong - I put the free functions in `dealii::` instead of `dealii::Simplex`.

Reply:
Overloading works correctly if they are in `dealii::Simplex` so I'll get rid of this.